README.md

```diff
@@ -114,9 +118,37 @@ BENCHMARK(BM_StringCopy);
 BENCHMARK_MAIN();
 ```
 
-Don't forget to inform your linker to add benchmark library e.g. through `-lbenchmark` compilation flag.
+Don't forget to inform your linker to link to the benchmark library e.g. through `-lbenchmark` compilation flag, and also the pthread library ('-lpthread'). If running google benchmarks on Windows, the shlwapi library must additionally be linked to (`-lshlwapi`) and the library location specified eg. ../../libraries/google_benchmark/lib. If running Solaris, kstat (`-lkstat`) must be linked to.
+
+The resultant compiled program should, if correctly compiled and linked, report timing for any registered functions with code within the `for(auto _ : state)` loops.
+
+
+### Basic usage code explanation
+
+The following code:
+```BENCHMARK(BM_StringCreation);```
+registers the function as a google benchmark function, to be run when the:
+```BENCHMARK_MAIN();```
+function is called, along with any other registered benchmark functions.
+
+When the benchmark is run, google benchmark first runs it a certain number of times to ascertain how long the function usually takes to run. This serves two purposes: (a) "warming up" the cpu cache/prediction-code by running the code a number of times, (b) finding out how many iterations of the function will be necessary to get a statistically-meaningful result. For example, a function that only takes a couple of nanoseconds to run is far more sensitive during measurement to the effects of other things the OS is doing (networking, background processes) than a function which takes a couple of seconds. So the 2-nanosecond function needs to be run a greater number of times to get accurate timings.
```

Comment:
Please linewrap and use proper lists

Reply:
When you say 'linewrap', what do you mean exactly.

Reply:
I'm not changing this commit.

Reply:
Right now line 134 is a long line. Please wrap at the same width as the rest to make it more readable.

Reply:
Why is this important when the HTML will wrap anyway?
Happy to split up the lining to space it out a bit.

Reply:
Not everyone reads it as HTML.